### ISSUE TITLE
Version 3.0

### ISSUE THREAD

**Issue by cristianoc72** [phootwork/phootwork#63]
I'ld like to create a `3.0` branch to start the development of the new phootwork major version.
The 3.0 version could be PHP 8+ because imho our library could benefit a lot from the new PHP 8 features, in particular union types and new string functions.

Ideas and suggestions are greatly wellcome! :smile:

**Comment by gossi** [comment]
yes absolutely. I'll tried this very naivly with some changes that weren't enough. New major sounds good, too. But I think v3 is enough ;)

**Comment by cristianoc72** [comment]
> But I think v3 is enough ;)

oh, of course! I don't know why, but when I was writing I thought that the last release was 3.01 :rofl:

**Comment by cristianoc72** [comment]
3.0 branch is here. The code is PHP8+, tests are green and Psalm is ok (except for 1 error but it's a Psalm issue with PHP8).
`php-cs-fixer` is not 100% PHP8 compatible by now, in particular for union types: since it doesn't recognize them, it force 1 space around `|` token. I.e.

```php
<?php declare(strict_types=1);

    //Correct style
    public function doSomething(array|string $action): mixed
    {
        .....................
    }

    //After php-cs-fixer fix
    public function doSomething(array | string $action): mixed
    {
        .....................
    }
```
When https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4702 issue will be completed, php-cs-fixer should correctly write union types (in alphabetic order, too :smile: )

**Comment by cristianoc72** [comment]
@gossi I realized a moment ago that I merged PR #67 into `3.0` branch: are you agree if I merge it into master from command line and push it by hand?

**Comment by gossi** [comment]
do whatever is needed 😂
